### PR TITLE
Fix autolink script

### PR DESCRIPTION
**Pull request by jinshin1013**
This PR fixes the autolink script including:
- Specifying the minSdkVersion for Android. Closes https://github.com/wix/react-native-navigation/issues/5983.
- Removing the RNN Pod added by the react-native link script.

@guyca  I've assumed the minSdkVersion will be 19 unless user specified higher. I've got the version from the doc, should this be updated to something higher?

### CODE REVIEW

File: autolink/postlink/gradleLinker.js [hunk 1]
```diff
@@ -69,6 +88,21 @@ class GradleLinker {
     return `"${DEFAULT_KOTLIN_VERSION}"`;
   }
 
+  /**
+   * Get the minSdkVersion value.
+   * @param { string } contents
+   */
+  _getMinSdkVersion(contents) {
+    var minSdkVersion = contents.match(/minSdkVersion = (\d*)/)
```
Comment (guyca) [inline autolink/postlink/gradleLinker.js]: We should account for zero or more spaces around to `=` symbol.
```suggestion
    var minSdkVersion = contents.match(/minSdkVersion\s=\s(\d*)/)
```

File: autolink/postlink/gradleLinker.js [hunk 2]
```diff
@@ -54,6 +57,22 @@ class GradleLinker {
...
+  _setMinSdkVersion(contents) {
+    var minSdkVersion = this._getMinSdkVersion(contents)
+    // If user entered minSdkVersion is lower than the default, set it to default.
+    if (minSdkVersion < DEFAULT_MIN_SDK_VERSION) {
+      debugn(`   Updating minSdkVersion to ${DEFAULT_MIN_SDK_VERSION}`)
+      return contents.replace(/minSdkVersion = \d*/, `minSdkVersion = ${DEFAULT_MIN_SDK_VERSION}`)
+    } 
+
+    debugn(`   Already specified minSdkVersion ${minSdkVersion}`)
+    return contents.replace(/minSdkVersion = \d*/, `minSdkVersion = ${minSdkVersion}`)
```
Comment (guyca) [inline autolink/postlink/gradleLinker.js]: lets account for white spaces around the equals sign
```suggestion
    return contents.replace(/minSdkVersion\s=\s\d*/, `minSdkVersion = ${minSdkVersion}`)
```

Comment (jinshin1013) [reply autolink/postlink/gradleLinker.js]: Nice catch! Had to use `\s{0,}=\s{0,}` instead.

File: autolink/postlink/gradleLinker.js (hunk 1 above)
Comment (jinshin1013) [reply autolink/postlink/gradleLinker.js]: Had to use `\s{0,}=\s{0,}` instead.

**Comment by jinshin1013** [comment]
@guyca Thanks for catching the regex issue, I'm still trying to get myself used to it 😓

> minSdk should be 19 as we don't support 18 and below.

Makes sense for minSdkVersion to be at least 19 and I also reckon most people would use 21 anyway. 

> Can you please explain why the change to podfile is needed?

I've found that whenever I run `react-native link react-native-navigation` it adds the pod definition for react-native-navigation. I'm using `@react-native-community/cli@3.2.0` so I don't think it's the cli problem. You should be able to reproduce this problem with fresh react-native project. Also as you mentioned I reckon it'd be good for the existing users who followed the previous installation step.

**Comment by guyca** [comment]
> I've found that whenever I run `react-native link react-native-navigation` it adds the pod definition for react-native-navigation. 

Isn't this desired? I thought `react-native link` adds all native dependencies to the podfile.

File: autolink/postlink/gradleLinker.js [hunk 3]
```diff
@@ -54,6 +57,22 @@ class GradleLinker {
...
+  /**
+   * Check the current minSdkVersion specified and if it's lower than
+   * the required version, set it to the required version otherwise leave as it is.
+   */
+  _setMinSdkVersion(contents) {
+    var minSdkVersion = this._getMinSdkVersion(contents)
+    // If user entered minSdkVersion is lower than the default, set it to default.
+    if (minSdkVersion < DEFAULT_MIN_SDK_VERSION) {
+      debugn(`   Updating minSdkVersion to ${DEFAULT_MIN_SDK_VERSION}`)
+      return contents.replace(/minSdkVersion\s{0,}=\s{0,}\d*/, `minSdkVersion = ${DEFAULT_MIN_SDK_VERSION}`)
```
Comment (guyca) [inline autolink/postlink/gradleLinker.js]: You can use `\s*` to match zero or more occurrences of non-breaking white spaces. Up to you :)

**Comment by jinshin1013** [comment]
> Isn't this desired? I thought react-native link adds all native dependencies to the podfile.

That is true for react-native version below 0.60. But I believe tools should assume users are on the "new" react-native and add somewhere in the documentation for the legacy linking (adding Pod definition). Thoughts?

File: autolink/postlink/gradleLinker.js (hunk 3 above)
Comment (jinshin1013) [reply autolink/postlink/gradleLinker.js]: Ah I see! Didn't even think about that hahaha. I'm off of my laptop so will probably just going to go with the current to save myself changing it :)

Comment (guyca) [reply autolink/postlink/gradleLinker.js]: No problem 😁

**Pull request merged**